includes/class-wc-payments-order-success-page.php

```diff
@@ -28,39 +28,74 @@ public function __construct() {
 	 */
 	public function register_payment_method_override() {
 		// Override the payment method title on the order received page.
-		add_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woopay_payment_method_name' ], 10, 2 );
+		add_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woocommerce_payments_payment_method_name' ], 10, 2 );
 	}
 
 	/**
 	 * Remove the hook to override the payment method name on the order received page before the order summary.
 	 */
 	public function unregister_payment_method_override() {
-		remove_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woopay_payment_method_name' ], 10 );
+		remove_filter( 'woocommerce_order_get_payment_method_title', [ $this, 'show_woocommerce_payments_payment_method_name' ], 10 );
 	}
 
 	/**
-	 * Add the WooPay logo and the last 4 digits of the card used to the payment method name
-	 * on the order received page.
+	 * Hooked into `woocommerce_order_get_payment_method_title` to change the payment method title on the
+	 * order received page for WooPay and BNPL orders.
 	 *
-	 * @param string            $payment_method_title the default payment method title.
-	 * @param WC_Abstract_Order $abstract_order the order being shown.
+	 * @param string $payment_method_title
+	 * @param WC_Abstract_Order $abstract_order
+	 * @return string
 	 */
-	public function show_woopay_payment_method_name( $payment_method_title, $abstract_order ) {
-
+	public function show_woocommerce_payments_payment_method_name( $payment_method_title, $abstract_order ) {
 		// Only change the payment method title on the order received page.
 		if ( ! is_order_received_page() ) {
 			return $payment_method_title;
 		}
 
-		$order_id = $abstract_order->get_id();
-		$order    = wc_get_order( $order_id );
-		if ( ! $order || ! $order->get_meta( 'is_woopay' ) ) {
+		$order_id          = $abstract_order->get_id();
+		$order             = wc_get_order( $order_id );
+		$payment_method_id = $order->get_payment_method();
+
+		if ( stripos( $payment_method_id, 'woocommerce_payments' ) !== 0 ) {
 			return $payment_method_title;
 		}
 
+		if ( ! $order ) {
+			return $payment_method_title;
+		}
+
+		// If this is a WooPay order, return the html for the WooPay payment method name.
+		if ( $order->get_meta( 'is_woopay' ) ) {
+			return $this->show_woopay_payment_method_name( $order );
+		}
+
+		$gateway           = WC()->payment_gateways()->payment_gateways()[ $payment_method_id ];
+		$payment_method    = $gateway->get_payment_method( $order );
+
+		// If this is a BNPL order, return the html for the BNPL payment method name.
+		if ( $payment_method->is_bnpl() ) {
+			$bnpl_output = $this->show_bnpl_payment_method_name( $gateway, $payment_method );
+
+			if ( $bnpl_output !== false ) {
+				return $bnpl_output;
+			}
+		}
```

Comment:
```suggestion
		// If this is a BNPL order, return the html for the BNPL payment method name.
		if ( $payment_method->is_bnpl() ) {
			$gateway                = WC()->payment_gateways()->payment_gateways()[ $payment_method_id ];
			$payment_method = $gateway->get_payment_method( $order );
			$bnpl_output          = $this->show_bnpl_payment_method_name( $gateway, $payment_method );

			if ( $bnpl_output !== false ) {
				return $bnpl_output;
			}
		}
```
I think we can move these two lines inside the conditional here. We only need the gateway and payment method if the order is a BNPL order AFAICT.

Reply:
Disregard, no we can't lol.